xds/internal/xdsclient/xds.go

```diff
@@ -561,6 +562,33 @@ func routesProtoToSlice(routes []*v3routepb.Route, logger *grpclog.PrefixLogger,
 	return routesRet, nil
 }
 
+func hashPoliciesProtoToSlice(policies []*v3routepb.RouteAction_HashPolicy) []*HashPolicy {
+	var hashPoliciesRet []*HashPolicy
+	for _, p := range policies {
+		var policy HashPolicy
+		policy.Terminal = p.Terminal
+		switch p.GetPolicySpecifier().(type) {
+		case *v3routepb.RouteAction_HashPolicy_Header_:
+			policy.HashPolicyType = HashPolicyTypeHeader
+			policy.HeaderName = p.GetHeader().HeaderName
+			policy.Regex = p.GetHeader().GetRegexRewrite().GetPattern().Regex
+			policy.RegexSubstitution = p.GetHeader().GetRegexRewrite().Substitution
+		// ConnectionProperties in Envoy specifies to hash the source ip
```

Comment:
Move this under `case *v3routepb.RouteAction_HashPolicy_ConnectionProperties_`.

Or, move this to the next config selector PR. This comment doesn't make much sense here.

Reply:
How does this not make sense?

Reply:
This comment would make more sense in config selector, where you hash the pointer. You don't need to explain here how it will be handled.